### PR TITLE
[DMC] Implement Jared Carthalion

### PR DESCRIPTION
**Pull request by PurpleCrowbar**

### CODE REVIEW

**Comment by PurpleCrowbar** [comment]
[[Jared Carthalion]]

File: Mage.Sets/src/mage/cards/j/JaredCarthalion.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        for (UUID targetId : getTargetPointer().getTargets(game, source)) {
+            Permanent creature = game.getPermanent(targetId);
+            if (creature != null) {
+                Counter counter;
+                counter = CounterType.P1P1.createInstance(creature.getColor().getColorCount());
+                creature.addCounters(counter, source, game);
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/j/JaredCarthalion.java:95]: you don’t need to declare the counter variable on a separate line, also you should get the color count, check if it’s positive, and then add the counters

File: Mage.Sets/src/mage/cards/j/JaredCarthalion.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    public JaredCarthalionUltimateEffect copy() {
+        return new JaredCarthalionUltimateEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller != null) {
+            Card card = game.getCard(targetPointer.getFirst(game, source));
+            if (card != null) {
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/j/JaredCarthalion.java:124]: you should combine the card and player check and invert them to flatten the code here

**Review by theelk801 — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/j/JaredCarthalion.java (hunk 1 above)
Comment (PurpleCrowbar) [reply Mage.Sets/src/mage/cards/j/JaredCarthalion.java:95]: Is it possible for a card to have a negative number of colors?

Comment (theelk801) [reply Mage.Sets/src/mage/cards/j/JaredCarthalion.java:95]: no but it can have zero

Comment (PurpleCrowbar) [reply Mage.Sets/src/mage/cards/j/JaredCarthalion.java:95]: In testing with a colorless creature (Solemn Simulacrum), the effect resolves as expected without error and without the creature receiving any counters. Do you think I should change it anyway or?

Comment (theelk801) [reply Mage.Sets/src/mage/cards/j/JaredCarthalion.java:95]: it should work either way, I'd just rather short circuit creating the counters and calling the method

**Pull request merged**